eng/common/TestResources/TestResources-Helpers.ps1

```diff
@@ -203,19 +230,40 @@ function SetDeploymentOutputs(
     $deploymentOutputs = BuildDeploymentOutputs $serviceName $azContext $deployment $deploymentEnvironmentVariables
 
     if ($OutFile) {
-        if (!$IsWindows) {
-            Write-Host 'File option is supported only on Windows'
-        }
+        if ($IsWindows -and $Language -eq 'dotnet') {
+            $outputFile = "$($templateFile.originalFilePath).env"
+
+            $environmentText = $deploymentOutputs | ConvertTo-Json;
+            $bytes = [System.Text.Encoding]::UTF8.GetBytes($environmentText)
+            $protectedBytes = [Security.Cryptography.ProtectedData]::Protect($bytes, $null, [Security.Cryptography.DataProtectionScope]::CurrentUser)
 
-        $outputFile = "$($templateFile.originalFilePath).env"
+            Set-Content $outputFile -Value $protectedBytes -AsByteStream -Force
+
+            Write-Host "Test environment settings`n$environmentText`nstored into encrypted $outputFile"
+        }
+        else {
```

Comment:
Don't you need to check SupportsDotenv?

Reply:
No, because at this point the user explicitly passed `-OutFile` or we flipped it based on the previous check of `$SupportsDotenv`. The only reason this inner check exists is to - for now - keep supporting DPAPI use for the .NET repo on Windows; otherwise, it'd just be this block (and will be eventually) under the `if ($OutFile) {` check.

Reply:
Yes I missed the OutFile is now holding the setting.